plugins/out_azure/azure.c

```diff
@@ -95,9 +98,27 @@ static int azure_format(const void *in_buf, size_t in_bytes,
         /* Append the time key */
         msgpack_pack_str(&mp_pck, flb_sds_len(ctx->time_key));
         msgpack_pack_str_body(&mp_pck,
-                              ctx->time_key,
-                              flb_sds_len(ctx->time_key));
-        msgpack_pack_double(&mp_pck, t);
+                            ctx->time_key,
+                            flb_sds_len(ctx->time_key));
+
+        if (ctx->time_generated == FLB_TRUE) {
+            /* Append the time value as ISO 8601 */
+            gmtime_r(&tm.tm.tv_sec, &tms);
+            s = strftime(time_formatted, sizeof(time_formatted) - 1,
+                            FLB_PACK_JSON_DATE_ISO8601_FMT, &tms);
+
+            len = snprintf(time_formatted + s,
+                            sizeof(time_formatted) - 1 - s,
+                            ".%06" PRIu64 "Z",
+                            (uint64_t) tm.tm.tv_nsec / 1000);
```

Comment:
Does the azure API support microsecond ?

https://docs.microsoft.com/en-us/azure/azure-monitor/logs/data-collector-api#request-headers
```
" The contents of the message field should follow the ISO 8601 format YYYY-MM-DDThh:mm:ssZ."
```

Reply:
Looks like they do. I've checked the timestamp from the log and the timestamp reported in Azure.

Timestamp in for e.g. docker: 2021-08-05T08:06:48.277911316Z
Timegenerated in Azure: 2021-08-05T08:06:48.277Z

I can't remember where I found the information but during implementation I've gathered that they support milliseconds.